data/lib/core/player.lua

```diff
@@ -534,3 +534,21 @@ function Player.sendBestiaryMilestoneReached(self, raceId)
 	msg:delete()
 	return true
 end
+
+function Player.updateClientExpGainRate(self)
+	-- Experience bonus (includes server rates)
+	local xpGainRate = 100 * Game.getExperienceStage(self:getLevel())
+	self:setBaseExpGain(xpGainRate)
+
+	-- Stamina bonus
+	local staminaMinutes = self:getStamina()
+	local staminaBonus = (staminaMinutes > 2340) and 150 or ((staminaMinutes < 840) and 50 or 100)
+	self:setStaminaExpBoost(staminaBonus)
+
+	-- Low level grinding bonus
+	-- TODO: Implement bonus system for low levels
+	local level = self:getLevel()
+	local levelBonus = (level > 1 and level <= 50) and 50 or 0
+	self:setGrindingExpBoost(levelBonus)
```

Comment:
Where does the name "grinding exp" come from?

Reply:
this bonus is the one with a level lower than 50, I couldn't implement it and I don't even have a better name for it, if you want to suggest something.

https://tibia.fandom.com/wiki/Experience_Point
![image](https://user-images.githubusercontent.com/26801045/235804186-fb993801-9287-4f4d-b090-a92c01a95a2e.png)